obs/project.go

```diff
@@ -40,13 +40,98 @@ type Project struct {
 	UseForBuild  *UseForBuild `json:"useForBuild,omitempty" xml:"useforbuild,omitempty"`
 }
 
-type Client struct {
-	Client   http.Client
+// OBS is the wrapper struct
+type OBS struct {
+	client  Client
+	options *Options
+}
+
+type obsClient struct {
+	*http.Client
+	*Options
+}
+
+// InvokeOBSEndpoint is the Interface method implementation over obsClient
+func (c *obsClient) InvokeOBSEndpoint(ctx context.Context, method string, url string, xmlData *bytes.Buffer) (*http.Response, error) {
+	//TODO implement me
+	req, err := http.NewRequestWithContext(ctx, method, url, xmlData)
+	if err != nil {
+		return nil, &APIError{
+			HTTPStatusCode: 0,
+			OBSStatusCode:  "",
+			Message:        fmt.Sprintf("creating obs project: creating request: %v", err),
+		}
+	}
+
+	req.SetBasicAuth(c.Username, c.Password)
+	req.Header.Set("Accept", "application/xml; charset=utf-8")
+
+	resp, err := c.InvokeOBSEndpoint(ctx, method, url, xmlData)
+	if err != nil {
+		return nil, &APIError{
+			HTTPStatusCode: 0,
+			OBSStatusCode:  "",
+			Message:        fmt.Sprintf("creating obs project: sending request: %v", err),
+		}
+	}
+
+	return resp, nil
+}
+
+type Client interface {
```

Comment:
```suggestion
// Client is an interface modeling supported OBS operations
type Client interface {
```

Reply:
Please also add counterfeiter annotations, see https://github.com/kubernetes-sigs/release-sdk/blob/1e0d8fcd5a1866844ba7d32b8c3915b2cca0429c/github/github.go#L75-L77 as an example

Reply:
Can we include it during our testing phase?